### PR TITLE
Reservations RP: updating to python sdk 0.2.1

### PR DESCRIPTION
**Pull request by juhee0202**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by juhee0202** [comment]
Pending on Python SDK 0.2.1 release

**Comment by juhee0202** [comment]
Hi @williexu I have prepared the Azure CLI PR in advance, but build will fail since the python SDK 0.2.1 is not yet released. As we had discussed, I am changing the CLI version to 0.2.1 as well with the minor enum changes.

File: src/command_modules/azure-cli-reservations/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.2.1
+++++++
+* minor fixes regarding enums.
```
Comment (williexu) [inline src/command_modules/azure-cli-reservations/HISTORY.rst]: What are the minor fixes?
Can you tell me what the new sdk changes consist of?

Comment (juhee0202) [reply src/command_modules/azure-cli-reservations/HISTORY.rst]: I had a comment from the REST API specs review that I should use string for “location”, "provisioningState' and all ARM defined attributes. Thus, I changed those two properties from enum to string.

Comment (juhee0202) [reply src/command_modules/azure-cli-reservations/HISTORY.rst]: https://github.com/Azure/azure-rest-api-specs/pull/3235/files

Comment (williexu) [reply src/command_modules/azure-cli-reservations/HISTORY.rst]: Sounds fine, thanks

**Pull request merged**